website/content/docs/connect/observability/grafanadashboards/consuldataplanedashboard.mdx

```diff
@@ -0,0 +1,91 @@
+page_title: Dashboard for Consul dataplane metrics
+description: >-
+  This Grafana dashboard that provides Consul dataplane metrics on Kubernetes deployments. Learn about the Grafana queries that produce the metrics and visualizations in this dashboard.
+---
+
+# Consul dataplane monitoring dashboard
+
+This page provides reference information about the Grafana dashboard configuration included in [this GitHub repository](https://github.com/YasminLorinKaygalak/GrafanaDemo/tree/main). The Consul dataplane dashboard provides a comprehensive view of the service health, performance, and resource utilization within the Consul service mesh. 
+
+You can monitor key metrics at both the cluster and service levels with this dashboard. It can help you ensure service reliability and performance.
+
+![Preview of the Consul dataplane dashboard](../../../../public/img/grafana/consul-dataplane-dashboard.png)
+
+## Consul dataplane metrics
+
+The Consul dataplane dashboard provides the following information about service mesh operations.
+
+### Live service count
+
+- **Grafana query:** `sum(envoy_server_live{app=~"$service"})`
+- **Description:** Displays the total number of live Envoy proxies currently running in the service mesh. It helps track the overall availability of services and identify any outages or other widespread issues in the service mesh.
+
+### Total request success rate
+
+- **Grafana query:** `sum(irate(envoy_cluster_upstream_rq_xx{...}[10m]))`
+- **Description:** Tracks the percentage of successful requests across the service mesh. It excludes 4xx and 5xx response codes to focus on operational success. Use it to monitor the overall reliability of your services.
+
+### Total failed requests
+
+- **Grafana query:** `sum(increase(envoy_cluster_upstream_rq_xx{...}[10m]))`
+- **Description:** This pie chart shows the total number of failed requests within the service mesh, categorized by service. It provides a visual breakdown of where failures are occurring, allowing operators to focus on problematic services.
+
+### Requests per second
+
+- **Grafana query:** `sum(rate(envoy_http_downstream_rq_total{...}[5m]))`
+- **Description:** This metric shows the rate of incoming HTTP requests per second to the selected services. It helps operators understand the current load on services and how much traffic they are processing.
+
+### Unhealthy clusters
+
+- **Grafana query:** `(sum(envoy_cluster_membership_healthy{...}) - sum(envoy_cluster_membership_total{...}))`
+- **Description:** This metric tracks the number of unhealthy clusters in the mesh, helping operators identify services that are experiencing issues and need attention to ensure operational health.
+
+### Heap size
+
+- **Grafana query:** `SUM(envoy_server_memory_heap_size{app=~"$service"})`
+- **Description:** This metric displays the total memory heap size of the Envoy proxies. Monitoring heap size is essential to detect memory issues and ensure that services are operating efficiently.
```

Comment:
```suggestion
- **Description:** This metric displays the total memory heap size of the Envoy proxies.
```
I think this last part should be omitted because the heap size is really dependent on how the proxy is configure, the number of connections it is processing, etc. Proxies that are processing high volumes of traffic may have larger heap sizes than proxies that are fronting lower traffic services. The mere presence of a large heap size alone is not indicative of a problem. Operators would need to evaluate other metrics to determine if the heap allocation is unusual given the traffic load and configuration profile of a given proxy.